test/functional/rpc_generateblock.py

```diff
@@ -51,37 +88,44 @@ def run_test(self):
         assert_equal(len(block['tx']), 1)
         assert_equal(block['tx'][0]['vout'][0]['scriptPubKey']['addresses'][0], combo_address)
 
-        # Generate 110 blocks to spend
-        node.generatetoaddress(110, address)
+        # Generate some blocks to spend
+        miniwallet.generate(10)
+        node.generate(100)
 
         # Generate some extra mempool transactions to verify they don't get mined
         for _ in range(10):
-            node.sendtoaddress(address, 0.001)
+            miniwallet.send_self_transfer(from_node=node)
 
         self.log.info('Generate block with txid')
-        txid = node.sendtoaddress(address, 1)
+        utxo = utxos.pop(0)
+        rawtx = create_miniwallet_rawtx(utxo=utxo, scriptPubKey=scriptPubKey)
+        txid = node.sendrawtransaction(rawtx)
         hash = node.generateblock(address, [txid])['hash']
-        block = node.getblock(hash, 1)
+        block = node.getblock(hash, 2)
+        tx = block['tx'][0]
+        utxo = {'txid': block['tx'][1]['txid'], 'vout': 0, 'value': block['tx'][1]['vout'][0]['value']}
+        utxos.append(utxo)
```

Comment:
If you did miniwallet.send_self_transfer(...) as suggested above, you could get rid of this. Again, if you want the utxo in the list of utxos you are maintaining, you could do: utxos.append(miniwallet.get_utxo())